### PR TITLE
fix: refactor sign.js and related tests

### PR DESCRIPTION
**Pull request by shrugs**
# 🚀 Description

I was working on related things and this file was really bothering me, so I refactored it and the related tests.

- [x] 📘 I've reviewed the [OpenZeppelin Contributor Guidelines](../blob/master/CONTRIBUTING.md)
- [x] ✅ I've added tests where applicable to test my new functionality.
- [x] 📖 I've made sure that my contracts are well-documented.
- [x] 🎨 I've run the JS/Solidity linters and fixed any issues (`npm run lint:all:fix`).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -55,5 +55,6 @@
     "truffle": "^4.1.13",
     "truffle-hdwallet-provider": "0.0.5",
     "web3-utils": "^1.0.0-beta.34"
-  }
+  },
+  "dependencies": {}
```
Comment (nventuro) [inline package.json]: I take it this and commiting the `package-lock.json` isn't intentional?

File: test/library/ECRecovery.test.js [hunk 2]
```diff
@@ -1,72 +1,66 @@
...
     const signer = '0x1e318623ab09fe6de3c9b8672098464aeda9100e';
-    const message = web3.sha3(TEST_MESSAGE);
     // eslint-disable-next-line max-len
     const signature = '0x331fe75a821c982f9127538858900d87d3ec1f9f737338ad67cad133fa48feff48e6fa0c18abc62e42820f05943e47af3e9fbe306ce74d64094bdf1691ee53e001';
-    (await ecrecovery.recover(message, signature)).should.equal(signer);
+    (await this.mock.recover(TEST_MESSAGE, signature)).should.equal(signer);
   });
 
   it('recover using web3.eth.sign()', async function () {
-    // Create the signature using account[0]
```
Comment (nventuro) [inline test/library/ECRecovery.test.js]: Were these `account[0]` comments inaccurate?

Comment (shrugs) [reply test/library/ECRecovery.test.js]: yeah they were using accounts[1] after the `[_, anyone]` change

File: package.json (hunk 1 above)
Comment (shrugs) [reply package.json]: I meant to remove the `ethereumjs-utils` package that we don't need anymore, next commit will do that correctly

**Comment by shrugs** [comment]
I was working on using SignatureBouncer for a side project and something wasn't working, so I started to debug. This file had a lot of clarity improvements to be made and some minor best practices stuff, so I went ahead and fixed them. That's about it tbh. I think they're worth including, especially because the `hashMessage` / `toEthSignedMessage` situation is much clearer.

**Pull request merged**